### ISSUE TITLE
Unexpected type widening from float to float | int in property setter

### ISSUE THREAD

**Issue by pydsigner** [microsoft/pyright#11716]
**Describe the bug**

I wrote a property setter that accepts either a float or a tuple of floats, and converts floats into tuples of floats. However, pyright introduces the possibility of an int, resulting in the property getter failing to check.

**Code or Screenshots**

```python
class X:
    @property
    def y(self) -> tuple[float, float]:
        return self._y
    @y.setter
    def y(self, value: float | tuple[float, float]):
        if isinstance(value, float):
            self._y = (value, value)
        else:
            self._y = value
```

Output:

```
❯ pyright test.py
/home/pydsigner/dev/test.py
  /home/pydsigner/dev/test.py:4:16 - error: Type "tuple[float, float] | int" is not assignable to return type "tuple[float, float]"
    Type "tuple[float, float] | int" is not assignable to type "tuple[float, float]"
      "int" is not assignable to "tuple[float, float]" (reportReturnType)
1 error, 0 warnings, 0 informations
```

**VS Code extension or command-line**

Tested with the extension and with command-line pyright 1.1.411 in Python 3.14.7.

**Comment by rchiodo** [comment]
This behavior is by design. The [Python typing specification](https://typing.python.org/en/latest/spec/special-types.html#special-cases-for-float-and-complex) explicitly allows an `int` argument wherever `float` is annotated. This does not convert the value to a float at runtime, and `isinstance(1, float)` is `False`.

As a result, the setter's annotation permits this:

```python
x = X()
x.y = 1
print(x.y)  # 1, not the tuple promised by the getter
```

The `else` branch can therefore receive `int | tuple[float, float]`, not just `tuple[float, float]`. Pyright is exposing an already-permitted possibility rather than unexpectedly widening the type. This behavior is not specific to properties.

The minimal fix is to handle both numeric types:

```python
@y.setter
def y(self, value: float | tuple[float, float]) -> None:
    if isinstance(value, (float, int)):
        self._y = (value, value)
    else:
        self._y = value
```

Integer elements are also acceptable in `tuple[float, float]` under the same typing rule. If actual float objects are needed for the scalar case, use `(float(value), float(value))` instead.

Closing as working as intended.
